src/test/kotlin/org/rust/ide/annotator/RsInvalidReexportErrorAnnotatorTest.kt

```diff
@@ -0,0 +1,156 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.annotator
+
+import org.rust.MockEdition
+import org.rust.cargo.project.workspace.CargoWorkspace
+
+@MockEdition(CargoWorkspace.Edition.EDITION_2018)
+class RsInvalidReexportErrorAnnotatorTest : RsAnnotatorTestBase(RsErrorAnnotator::class) {
+    fun `test reexport of public item from accessible module`() = checkErrors("""
+        mod foo {
+            mod bar {
+                pub struct Foo;
+                pub mod foo {}
+            }
+            pub use bar::Foo;
+            pub use bar::foo;
+        }
+    """)
+
+    fun `test E0603 reexport of public item from inaccessible module`() = checkErrors("""
+        mod foo {
+            mod bar {
+                mod baz {}
+            }
+            pub use <error descr="Module `bar::baz` is private [E0603]">bar::baz</error>;
+        }
+    """)
+
+    fun `test E0365 reexport of private module`() = checkErrors("""
+        mod foo {
+            mod bar {}
+            pub use <error descr="`bar` is private, and cannot be re-exported [E0365]">bar</error> as baz;
+        }
+    """)
+
+    fun `test E0365 star reexport of private module`() = checkErrors("""
+        mod foo {
+            mod bar {}
+            pub use bar::*;
+        }
+    """)
+
+    fun `test E0364 reexport of private item`() = checkErrors("""
+        mod foo {
+            struct Foo;
+            pub use <error descr="`Foo` is private, and cannot be re-exported [E0364]">Foo</error> as Bar;
+        }
+    """)
+
+    fun `test E0364 reexport of private item from parent module`() = checkErrors("""
+        mod foo {
+            fn foo() {}
+
+            mod bar {
+                pub use super::<error descr="`foo` is private, and cannot be re-exported [E0364]">foo</error>;
+            }
+        }
+    """)
+
+    fun `test reexport multiple namespaces`() = checkErrors("""
+        mod foo {
+            pub struct A { a: u32 }
+            const A: u32 = 0;
+
+            pub use A as B;
+        }
+        mod bar {
+            struct A { a: u32 }
+            pub const A: u32 = 0;
+
+            pub use A as B;
+        }
+    """)
```

Comment:
Let's add a similar test but when items have the same namespace. It should produce only [E0428]